examples/ethswctld/cetherswitch.cc

```diff
@@ -82,13 +82,14 @@ cetherswitch::cetherswitch(
 
 cetherswitch::~cetherswitch() {}
 
-void cetherswitch::handle_timeout(cthread &thread, uint32_t timer_id) {
+void cetherswitch::handle_timeout(void *userdata) {
   try {
+    int timer_id = (long)userdata;
```

Comment:
weird but necessary. Looking at this right now I think it should be a more c++ style cast :)